Geometry/GEMGeometryBuilder/src/GEMGeometryBuilder.cc

```diff
@@ -75,14 +77,22 @@ void GEMGeometryBuilder::build(GEMGeometry& theGeometry,
     fv.firstChild();
     fv.firstChild();
 
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("Geometry") << "MuonGeometry 1 " << fv.geoHistory() << " Levels "
+                                 << mdddnum.geoHistoryToBaseNumber(fv.geoHistory()).getLevels();
+#endif
     int rawidCh = gemNum.baseNumberToUnitNumber(mdddnum.geoHistoryToBaseNumber(fv.geoHistory()));
     GEMDetId detIdCh = GEMDetId(rawidCh);
 
     // back to chambers
 
     fv.parent();
     fv.parent();
-
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("Geometry") << "MuonGeometry 2 " << fv.geoHistory() << " Levels "
+                                 << mdddnum.geoHistoryToBaseNumber(fv.geoHistory()).getLevels();
+    gemNum.baseNumberToUnitNumber(mdddnum.geoHistoryToBaseNumber(fv.geoHistory()));
```

Comment:
This line looks dangerous -- a non-const method called within an #ifdef. A reviewer of the code is prone to wonder if the code will behave differently depending on whether or not it was compiled with the EDM_ML_DEBUG flag. In this case, the method is effectively const even though it is not declared that way, and it is being used here to output additional debug messages. It would be nice if it were possible to declare `gemNum` const.
At the very least, a comment is needed here to explain the call and say it doesn't change behavior.

Reply:
Changed the method to const

Reply:
Can `gemNum` on line 74 now be declared const? That would show that its methods can be called within #ifdef blocks without causing changes.